### ISSUE TITLE
`to_json` produces invalid json (keys not wrapped in double quotes) against Unions of struct column

### ISSUE THREAD

**Issue by Swoorup** [duckdb/duckdb#5583]
### What happens?

When using a Union of struct column, using `to_json` operation on a select statement appears to return an invalid json. The json keys are wrapping in single quotes instead of double quotes.

### To Reproduce

1. Fire up duckdb CLI.
2. Paste the code below.
  ```sql
create table if not exists event(
    payload union(
        trade struct( 
            trade_id UBigInt, 
            price Decimal(18, 6),  
            volume Decimal(18, 6), 
            is_buyer_maker bool 
        ),
        l2 struct(
            sequence UBigInt,
            is_snapshot bool,
            bids struct(price Decimal(18, 6), volume Decimal(18, 6))[], 
            asks struct(price Decimal(18, 6), volume Decimal(18, 6))[]
        )
    )
);
insert into event(
    payload
) values (
    union_value(l2 := {
        sequence: 1231231, 
        is_snapshot: true,
        bids: [ { price: 12, volume: 32} ], 
        asks: []
    })
);

insert into event(
    payload
) values (
    union_value(trade := struct_pack( 
        trade_id := 1231231, 
        price := 97.8, 
        volume := 5, 
        is_buyer_maker := true 
    ))
);
  ```
3. Run the query which retrieves the json for the union column: 
   ```sql
    from event select to_json(payload);
    ```
5. Observe the result set of the last query.
```shell
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                               to_json(payload)                                                │
│                                                     json                                                      │
├───────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ "{'sequence': 1231231, 'is_snapshot': true, 'bids': [{'price': 12.000000, 'volume': 32.000000}], 'asks': []}" │
│ "{'trade_id': 1231231, 'price': 97.800000, 'volume': 5.000000, 'is_buyer_maker': true}"                       │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

### OS:

aarch

### DuckDB Version:

v0.6.0 2213f9c946

### DuckDB Client:

CLI

### Full Name:

Swoorup Joshi

### Affiliation:

N/A

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Tishj** [comment]
seems like it needs special handling in `json_create.cpp:GetJSONType`

```c++
	case LogicalTypeId::STRUCT: {
		child_list_t<LogicalType> child_types;
		for (const auto &child_type : StructType::GetChildTypes(type)) {
			const_struct_names[child_type.first] = make_unique<Vector>(Value(child_type.first));
			child_types.emplace_back(child_type.first, GetJSONType(const_struct_names, child_type.second));
		}
		return LogicalType::STRUCT(child_types);
	}
	case LogicalTypeId::MAP: {
		return LogicalType::MAP(LogicalType::VARCHAR, GetJSONType(const_struct_names, MapType::ValueType(type)));
	}
	// All other types (e.g. date) are cast to VARCHAR
	default:
		return LogicalTypeId::VARCHAR;
	}
}
```

**Comment by Maxxen** [comment]
This is fixed in #5617. The example now produces:

```
D  from event select to_json(payload);
┌────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                        to_json(payload)                                        │
│                                              json                                              │
├────────────────────────────────────────────────────────────────────────────────────────────────┤
│ {"l2":{"sequence":1231231,"is_snapshot":true,"bids":[{"price":12.0,"volume":32.0}],"asks":[]}} │
│ {"trade":{"trade_id":1231231,"price":97.8,"volume":5.0,"is_buyer_maker":true}}                 │
└────────────────────────────────────────────────────────────────────────────────────────────────┘
```
